src/map/battlefield.cpp

```diff
@@ -453,11 +540,13 @@ void CBattlefield::OpenChestinBcnm() {
 
 bool CBattlefield::loseBcnm() {
     beforeCleanup();
-    for (auto&& PChar : m_PlayerList)
+    for (int i = 0; i < m_PlayerList.size(); i++)
```

Comment:
I don't think this is right, but it's something I can look at (it probably should have just had its own issue though)